### PR TITLE
Dockerize gazzetteer db

### PR DESCRIPTION
**Pull request by strk**
TODO:

  - [x] Make installation of schema work
  - [x] Put under CI testing

### CODE REVIEW

**Comment by palmerj** [comment]
Wow I'm surprised there is a docker image for 8.4. What about PostGIS 1.5 (GEO 3.2, PROJ=4.6)?

**Comment by strk** [comment]
Eh, that part I'm going to find out next (or will build from source).
The INSTALLATION file did not mention that requirement, I'm going to send some side-PRs to improve both documentation and installation scripts, would simplify this task

**Comment by strk** [comment]
Ok, PostGIS is built from source. Database schema is created. Only it does NOT currently include the `gazetteer_web` schema, should it @SPlanzer ?

**Comment by strk** [comment]
NOTE: as of 6a88e37 the database is created with 2 users `gaz_web` and `gaz_web_dev` both with password `********`, which is unlikely what we want.

**Comment by SPlanzer** [comment]
> Ok, PostGIS is built from source. Database schema is created. Only it does NOT currently include the gazetteer_web schema, should it @SPlanzer ?

gazetteer_web schema is most likely not required for testing the porting. I am happy to approve this pull request for this reason, however it would be useful to include the schema to support any future development of the application. 

 if `%psql -f gazetteer_web_schema.sql || exit 1` was added to `src/install.sh` prior to `$psql -f gazetteer_grant.sql  || exit 1` the gazetteer_web schema could also be incuded

https://github.com/linz/gazetteer/blob/615d7ed9ca3e40bb13900f75842425e5893b8274/src/sql/install.sh#L60-L62

**Comment by SPlanzer** [comment]
> NOTE: as of 6a88e37 the database is created with 2 users gaz_web and gaz_web_dev both with password ********, which is unlikely what we want.

password set in following files
```
sql/Makefile:	PGPASSWORD=******** psql -h localhost -p 7432 -U gaz_web gazetteer
sql/gazetteer_roles.sql:CREATE USER gaz_web IN ROLE gaz_web_reader ENCRYPTED PASSWORD '********';
sql/gazetteer_roles.sql:CREATE USER gaz_web_dev IN ROLE gaz_web_reader, gaz_web_developer ENCRYPTED PASSWORD '********';
```

I will approve this for CI purposes. obviously this is not fine for prd deployment - prd deployment out of scope of this project

File: src/sql/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+# Install PostGIS 1.5.8 from source
+RUN curl -s -L \
+  https://ftp.osuosl.org/pub/osgeo/download/postgis/source/postgis-1.5.8.tar.gz | \
+  tar xz && cd postgis-1.5.8 && \
+  ./configure && make && make install && \
+  cd .. && rm postgis-1.5.8 -rf
+
+# Install gazeteer schema
+ADD *.sql build_gazetteer_export.py install.sh /sql/
```
Comment (SPlanzer) [inline src/sql/Dockerfile]: %psql% -f gazetteer_web_schema.sql  needs to be added to install.sh to deployed the gazetteer_web schema

Comment (strk) [reply src/sql/Dockerfile]: I had Dockerfile run that script outside of install.sh, see https://github.com/linz/gazetteer/pull/64/files#diff-cc0a840aa5e9e81d98c2b58dd2f7c98eR44 (install_web.sh sources gazetteer_web_schema.sql)

**Comment by strk** [comment]
Current problem is the circular dependency between scripts, as reported in https://github.com/linz/gazetteer/issues/69, which results in:

> psql:gazetteer_grant.sql:295: ERROR:  schema "gazetteer_web" does not exist

See:
https://github.com/linz/gazetteer/pull/64/checks?check_run_id=518947034#step:4:7700

I'm marking that issue as a blocker for this

**Comment by strk** [comment]
I've rebased this PR on #72 to get that fix

**Pull request merged**